platform/api/azure/instance.go

```diff
@@ -174,9 +177,18 @@ func (a *API) CreateInstance(name, userdata, sshkey, resourceGroup, storageAccou
 	}, nil
 }
 
-func (a *API) TerminateInstance(name, resourceGroup string) error {
-	_, err := a.compClient.Delete(resourceGroup, name, nil)
-	return err
+func (a *API) TerminateInstance(machine *Machine, resourceGroup string) error {
```

Comment:
Could you please add a function documentation comment as to what resources do and don't get terminated here?

Reply:
Thanks, done